tools/document-pii/src/main.rs

```diff
@@ -93,44 +85,62 @@ impl Cli {
             AstItemCollector::collect(&rust_file_paths)?
         };
 
-        let pii_types = match self.item.as_deref() {
-            // If user provides path to an item, find PII_fields under this item in particular.
-            Some(path) => types_and_use_statements.find_pii_fields_of_type(path, &self.pii_values),
-            // If no item is provided, find PII fields of all types in crate/workspace.
-            None => types_and_use_statements.find_pii_fields_of_all_types(&self.pii_values),
-        }?;
+        let pii_types =
+            types_and_use_statements.find_pii_fields(self.item.as_deref(), &self.pii_values)?;
 
         // Function also takes a string to replace unnamed fields, for now we just remove them.
-        let output_vec = get_pii_fields_output(pii_types, "".to_string());
+        let output_vec = Output::from_btreeset(pii_types);
 
         match self.output {
-            Some(ref path) => self.write_pii(File::create(path)?, &output_vec)?,
-            None => self.write_pii(std::io::stdout(), &output_vec)?,
-        }
+            Some(ref path) => serde_json::to_writer_pretty(File::create(path)?, &output_vec)?,
+            None => serde_json::to_writer_pretty(std::io::stdout(), &output_vec)?,
+        };
 
         Ok(())
     }
 }
 
-/// Represent the PII fields in a format that will be used in the final output.
-fn get_pii_fields_output(
-    pii_types: BTreeSet<Vec<TypeAndField>>,
-    unnamed_replace: String,
-) -> Vec<String> {
-    let mut output_vec = vec![];
-    for pii in pii_types {
-        let mut output = String::new();
-        output.push_str(&pii[0].qualified_type_name);
-
-        for path in pii {
-            output.push_str(&format!(".{}", path.field_ident));
+#[derive(Serialize, Default, Debug)]
+struct Output {
+    path: String,
+    additional_properties: bool,
+}
+
+impl Output {
+    fn new(pii_type: FieldsWithAttribute) -> Self {
+        let mut output = Self {
+            additional_properties: pii_type.attributes.contains_key("additional_properties"),
+            ..Default::default()
+        };
+
+        output
+            .path
+            .push_str(&pii_type.type_and_fields[0].qualified_type_name);
+
+        let mut iter = pii_type.type_and_fields.iter().peekable();
+        while let Some(path) = iter.next() {
+            // If field has attribute "additional_properties" it means it's not a real field
+            // but represents unstrucutred data. So we remove it and pass the information as a boolean
+            // in order to properly document this fact in the docs.
+            if !(output.additional_properties && iter.peek().is_none()) {
+                output.path.push_str(&format!(".{}", path.field_ident));
+            }
```

Comment:
Not adding anything to the path here could be problematic. If you have a struct like

```rust
struct Foo {
  #[metastructure(pii = "true")]
  outer: struct Outer {
    #[metastructure(additional_properties, retain = "true", pii ="true")]
    inner: Object<Value>,
  }
}
```

wouldn't this give you two identical paths?

```json
[
  {"path": "Foo.outer", "additional_properties": false},
  {"path": "Foo.outer", "additional_properties": true}
]
```

Reply:
Might not be a problem if the client of this output knows how to interpret the "additional_properties" flag.